pkg/controller/ambassadorinstallation/scout.go

```diff
@@ -0,0 +1,107 @@
+// From ambassador/internal/pkg/edgectl/scout.go with some modifications & simplifications.
+
+package ambassadorinstallation
+
+import (
+	"context"
+	"os"
+
+	"github.com/datawire/ambassador-operator/version"
+	"github.com/datawire/ambassador/pkg/metriton"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+// The Scout structure maintains an index, which is the count of calls
+// to Report, incrementing each call.  This provides a sequence of actions
+// to make it easier to search through the reports.
+// The Reporter is simply the Metriton reporting object.
+type Scout struct {
+	index    int
+	Reporter *metriton.Reporter
+}
+
+// Metadata is simply a key and an untyped value, instances passed in as parameters
+// to (s *Scout) Report
+type ScoutMeta struct {
+	Key   string
+	Value interface{}
+}
+
+// Function to get an installID, given a Reporter.  This is a standard
+// function form since we aren't allowed to add methods to an external type.
+func ThisInstallID(r *metriton.Reporter) (string, error) {
+	// Have cluster ID?
+	thisID := os.Getenv("AMBASSADOR_CLUSTER_ID")
```

Comment:
Who sets this `AMBASSADOR_CLUSTER_ID` value?

Reply:
this is a stale issue--see below.

Reply:
The `ThisInstallID` can then be deleted if unused.

Reply:
deleted